lib/clone.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+var Cesium = require('cesium');
+var ArrayStorage = require('./ArrayStorage');
+
+var defaultValue = Cesium.defaultValue;
+
+module.exports = clone;
+
+/**
+ * Clones an object, returning a new object containing the same properties.
+ * Modified from Cesium.clone to support typed arrays, buffers, and the ArrayStorage class.
+ *
+ * @param {Object} object The object to clone.
+ * @param {Boolean} [deep=false] If true, all properties will be deep cloned recursively.
+ * @returns {Object} The cloned object.
+ *
+ * @private
+ */
+function clone(object, deep) {
```

Comment:
This function appears to only be used in tests.  a) Do we really need it, and b) if yes we should move it to the `specs` directory.